database/connection.go

```diff
@@ -100,6 +105,26 @@ func (c *connection) Database() API {
 	return &api{db: &transactable{c.client}, configuration: c.configuration}
 }
 
+// NewBulkCreate API creating multiple records at the same time.
+func NewBulkCreate[T record.Record](cfg Configuration) (BulkCreate[T], error) {
```

Comment:
Why create a new connection type instead of just adding a new method on `database.API` that takes a slice of records and inserts them in a single statement?

Reply:
1. because then the caller would need to buffer the records
2. we want bulk action transactions open for a little a time as possible
3. we don't want bulk actions spread over multiple tables with other operations as this could lead to dead locks.
This accomplishes all three of those and only locks tables at the moment it does the insert and then closes the transaction.